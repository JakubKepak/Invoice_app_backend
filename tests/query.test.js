import { gql } from "apollo-server";
import { createTestServer } from "./helper";

const INVOICES = gql`
  {
    getInvoices {
      id
    }
  }
`;

describe("queries", () => {
  test("get invoices", async () => {
    const { query } = createTestServer({
      Invoice: {
        //   Mocking a find method of mongoose
        find: jest.fn(() => [{ id: 1 }, { id: 2 }]),
      },
    });

    const res = await query({ query: INVOICES });
    expect(res).toMatchSnapshot();
  });
});
