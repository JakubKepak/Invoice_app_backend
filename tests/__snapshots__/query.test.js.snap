// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries get invoices 1`] = `
Object {
  "data": Object {
    "getInvoices": Array [
      Object {
        "id": "1",
      },
      Object {
        "id": "2",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
